### PR TITLE
Refactor: point of sale service

### PR DESCRIPTION
**Pull request by JustSamuel**
# Description
Same story as the container and product service.

## Related issues/external references
#153 

## Types of changes
- Bug fix _(non-breaking change which fixes an issue)_

### CODE REVIEW

File: src/service/product-service.ts [hunk 1]
```diff
@@ -154,85 +135,63 @@ export function parseGetProductFilters(req: RequestWithToken): ProductFilterPara
...
-      vat,
+      vat: {
+        id: revision.vat.id,
+        percentage: revision.vat.percentage,
+        hidden: revision.vat.hidden,
+      },
     };
   }
 
+  private static  revisionSubQuery(revision?: number): string {
```
Comment (Yoronex) [inline src/service/product-service.ts:178]: Two spaces :(

File: test/unit/service/point-of-sale-service.ts [hunk 2]
```diff
@@ -252,7 +252,7 @@ describe('PointOfSaleService', async (): Promise<void> => {
         useAuthentication: true,
       };
 
-      const response = await PointOfSaleService.directPointOfSaleUpdate(update);
+      const response = (await PointOfSaleService.updatePointOfSale(update)) as PointOfSaleWithContainersResponse;
```
Comment (Yoronex) [inline test/unit/service/point-of-sale-service.ts:255]: I think the update function should have a default response (with or without containers/products) such that the responses are also always consistent.

**Review by Yoronex — CHANGES_REQUESTED** [review]

File: src/service/product-service.ts [hunk 3]
```diff
@@ -120,21 +106,16 @@ export interface ProductFilterParameters {
   priceList?: boolean;
 }
 
+// TODO figure out why this is unused
```
[the remark this answers is not in the document]
Comment (JustSamuel) [reply src/service/product-service.ts:109]: Nah, it is just never used, as in: nowhere does a controller expose the ability to use the filter params. However fixing this is outside the scope of this PR. 

I have updated the TODO.

File: src/service/point-of-sale-service.ts [hunk 4]
```diff
@@ -163,50 +128,23 @@ export default class PointOfSaleService {
...
-    return {
-      name: pointOfSale.name,
-      owner: parseUserToBaseResponse(pointOfSale.pointOfSale.owner, false),
-      id: pointOfSale.pointOfSale.id,
-      revision: pointOfSale.revision,
-    } as PointOfSaleResponse;
-  }
-
-  public static async applyPointOfSaleUpdate(base: PointOfSale, update: BasePointOfSaleParams) {
+  public static async updatePointOfSale(update: UpdatePointOfSaleParams) {
```
Comment (Yoronex) [inline src/service/point-of-sale-service.ts:146]: This function still has an `|` in its return type (`PointOfSaleResponse` or `PointOfSaleWithContainersResponse`). Please make explicit which type is returned, as this does not seem to be a choice for the user.

**Review by Yoronex — APPROVED** [review]

**Pull request merged**